### ISSUE TITLE
ReduceByKey does not work with topics with > 1 partition.

### ISSUE THREAD

**Issue by jozi-k** [confluentinc/examples#78]
I have simple application which takes input-topic, which uses 10 partitions. Output topic should produce aggregation of these records and has 2 partitions. See part of code:
```
case class User(name: String, sum: Integer)

val avro: KStream[String, GenericRecord] = builder.stream("input-topic")
val valuesMapped: KStream[String, Foo] = mco.mapValues(r => User(r))
val names: KStream[String, Integer] = valuesMapped.map { (_, u) => new KeyValue(u.name, u.sum)}
val agg: KTable[String, Integer] = names.reduceByKey((s1, s2) => s1 + s2, stringSerde, intSerde, "Aggregate")

agg.to(stringSerde, intSerde, "aggregate")
```
N.B. I added explicit serde classes to reduceByKey as explained in #73. When I start this application, I get following error:
```
org.apache.kafka.streams.errors.StreamsException: Store Aggregate's change log (aggregation-scala-example-Aggregate-changelog) does not contain partition 1
	at org.apache.kafka.streams.processor.internals.ProcessorStateManager.register(ProcessorStateManager.java:204)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.register(ProcessorContextImpl.java:116)
	at org.apache.kafka.streams.state.internals.RocksDBStore.init(RocksDBStore.java:202)
    ...

```
which is true, I see only aggregation-scala-example-Aggregate-changelog-0 in kafka-logs. There is no 1. It seems it is not possible to use kafka streams on topics with more than 1 partition.

**Comment by mjsax** [comment]
Can you please share the whole stack trace? Can you reset the application using `bin/kafka-streams-application-reset` and try again? If resetting does not resolve the problem, is seems to be a bug (I assume you do not create/modify this topic, but it is entirely managed by Kafka Streams).

**Comment by jozi-k** [comment]
You can view stack trace [here](https://github.com/confluentinc/examples/files/553269/error.txt). You are right, I do not create or modify topic mentioned in stack trace, it should be entirely managed by kafka streams. Resetting application have no effect, still get the same error.

**Comment by miguno** [comment]
Perhaps that's a bug that was fixed in 0.10.1 already?  (`reduceByKey` is 0.10.0.x)

**Comment by jozi-k** [comment]
Yes, using `groupByKey()` and `reduce()` with kafka-streams 0.10.1 resolved it. Thanks for hint @miguno , will try my other tests against new kafka.
